pkg/kontainerdrivermetadata/data.go

```diff
@@ -160,22 +167,40 @@ func (md *MetadataController) saveSystemImages(K8sVersionRKESystemImages map[str
 	return md.updateSettings(maxVersionForMajorK8sVersion, rancherVersion, ServiceOptions, DefaultK8sVersions, deprecatedMap)
 }
 
-func toIgnoreForAllK8s(rancherVersionInfo rketypes.K8sVersionInfo, rancherVersion string) bool {
-	if rancherVersionInfo.DeprecateRancherVersion != "" && mVersion.Compare(rancherVersion, rancherVersionInfo.DeprecateRancherVersion, ">=") {
-		return true
+func toIgnoreForAllK8s(info rketypes.K8sVersionInfo, rancherVersion string) bool {
+	rVersion, err := msemver.NewVersion(rancherVersion)
+	if err != nil {
+		return false
 	}
-	if rancherVersionInfo.MinRancherVersion != "" && mVersion.Compare(rancherVersion, rancherVersionInfo.MinRancherVersion, "<") {
-		// only respect min versions, even if max is present - we need to support upgraded clusters
```

Comment:
This comment should probably be retained in the updated code in the appropriate place